Playwright.Axe/NodeDependencies/src/axe/run-partial-and-merge.ts

```diff
@@ -0,0 +1,62 @@
+import type axe from "axe-core";
+import type { ElementContext, RunOptions, PartialResult } from "axe-core";
+
+type AxeType = typeof axe;
+
+/**
+ * Recursively calls Axe runPartial in each frame and adds it to an array.
+ */
+function runPartialRecursive(context: ElementContext, options: RunOptions, win = window) : Promise<PartialResult>[] {
+    const axe = win["axe"] as AxeType;
```

Comment:
I think we probably need to rethink this strategy for running `runPartial` recursively in each frame; it would be good to sync with the folks on the Deque side familiar with similar work in their other libraries. Specifically:

1. This code assumes free access to subframes' `window` properties, but that isn't available in the case where subframes have different origins than their parent frames
2. Even if this were technically feasible, we wouldn't want to marshal results from one frame under test to another like this; there might be a security boundary between the two frames such that it would be unsafe for us to process information from one frame's DOM in a different frame.

I think `runPartialRecursive` is something that this library should be implementing in the C# side within the playwright execution context, not as JavaScript that runs inside the page under test. Javascript within each frame can't necessarily run code in the context of all its child frames, but Playwright itself can. You can see an example of this distinction in the [`runPartialRecursive` implementation of `@axe-core/playwright`](https://github.com/dequelabs/axe-core-npm/blob/598ab440cfee90abc28bd73e90f1f33cad8225dc/packages/playwright/src/index.ts#L236), where `runPartialRecursive` runs within the context of the Playwright JavaScript context (not the page under test's).

There is also some other trickiness around execution order that `@axe-core/playwright` is handling which this implementation is currently omitting (see the comment banner above its implementation about ordering of injection vs scanning). Overall, I think it would be a good idea to take Deque up on their offer of collaborating on this work; there's a lot of subtlety to it that they've already given a lot of thought to in their similar libraries.